Plugin SDK - Host Emulator - phase 2

dmitry-yudakov opened the pull request:
Planned development:

- move MomentumRequired to HostEmulator and show automatically when on PROD
- separate world and space interfaces
- widgets panel
- use react-router

New command is added to `momentum-plugin` script - `start:plugin` - it runs the plugin with craco config and allows using it inside momentum. Currently it has problems with hot refresh and non-base roots after reload.

`momentum-plugin start` command (set as `start` script for plugin_miro) runs the dev server using react-scripts without custom craco staff - it works with hot refresh and reloads and generally is a more convenient options for plugin development in isolation.

dmitry-yudakov committed:
refactor(plugins): HostEmulator shows MomentumRequired or WorldEmulator depending on env

MomentumRequired page will be needed for all plugins and this logic is better to be in sdk rather than plugin

dmitry-yudakov committed:
feat(host-emulator): use react-router for different parts of the emulator

dmitry-yudakov committed:
prettier

dmitry-yudakov committed:
Merge branch 'develop-plugins' into feature/develop-plugins-uber-host-emulator

# Conflicts:
#	packages/sdk/package.json

PAKovalev commented on packages/sdk/src/emulator/components/WorldEmulator/WorldEmulator.tsx:
```diff
@@ -0,0 +1,38 @@
+      <Switch>
+        <Route exact path="/">
+          <div>
+            <button onClick={() => history.push(`/collaboration/${DUMMY_SPACE_ID}`)}>
```
Lets create routes as const.

PAKovalev commented on packages/sdk/src/emulator/components/SpaceEmulator/SpaceEmulator.tsx:
```diff
@@ -0,0 +1,55 @@
+        </styled.SpaceTopBar>
+        <styled.SpaceContent>
+          <Switch>
+            <Route path={`${path}/dashboard`}>
```
Lets create routes as const.

dmitry-yudakov committed:
Merge branch 'develop-plugins' into feature/develop-plugins-uber-host-emulator

# Conflicts:
#	packages/sdk/src/emulator/components/SpaceTabEmulator/SpaceTabEmulator.tsx

dmitry-yudakov committed:
Merge branch 'develop-plugins' into feature/develop-plugins-uber-host-emulator

dmitry-yudakov committed:
feat(sdk): start and start:plugin commands in momentum script

start command uses react-scripts for dev watch/build of plugin without using module federation - it helps with hot refresh and refreshing page on some url.
start:plugin command uses craco start with our config, that allows running in a mode allowing to plug into momentum process. We need to figure out how to make reloads working with routes.

dmitry-yudakov committed:
refactor(host-emulator): use ROUTES object instead of hardcoded paths

Daboo1998 commented on packages/sdk/bin/momentum-plugin.js:
```diff
@@ -1,50 +1,72 @@
+  case 'test': {
+    const nodeArgs = scriptIndex > 0 ? args.slice(0, scriptIndex) : [];
+    const configPath = require.resolve(`../craco.config.js`);
+    const command = script === 'start:plugin' ? 'start' : script;
```
instead of `start:plugin` I would make it a parameter you pass to something like `momentum-plugin start --no-craco` or `momentum-plugin start --standalone` or something else like that (maybe something shorter)

dmitry-yudakov replied:
It might mix with the underlying commands arguments - it's less ambiguous to use just just "command:specifier".
I also wanted the `start` command to work in standalone mode with hot refresh and so on - and `start:plugin` to be for the heavier and currently less functional version allowing to also plug it in local momentum.

Daboo1998 replied:
yes yes, I know I just don't like the `start:plugin` cause first of all we already have the word "plugin" in the script. I would suggest to look what arguments it can collide with and use a one that won't be colliding

Daboo1998 commented on packages/sdk/src/emulator/components/MomentumRequiredPage/MomentumRequiredPage.tsx:
```diff
@@ -1,15 +1,16 @@
-
   return (
     <styled.Container>
-      <Text size="l" text={t('messages.momentumRequired')} weight="bold" />
```
Why removed translation?

dmitry-yudakov replied:
To spare the need of having the translation file for SDK and make it simpler.

Daboo1998 commented on packages/sdk/src/emulator/components/SpaceEmulator/SpaceEmulator.tsx:
```diff
@@ -0,0 +1,59 @@
+  console.log('RENDER SpaceEmulator', {plugin, spaceId});
+  const history = useHistory();
+
+  const [topBar = <span />, setTopBar] = useState<JSX.Element | null>(null);
```
why `<span />` not as initial value in `useState` call?

Daboo1998 reviewed (approved):
I approve due to nature of the new setup
